omnibus/files/private-chef-ctl-commands/spec/secrets_spec.rb

```diff
@@ -0,0 +1,151 @@
+      }.to output("Please restart these services: bookshelf\n").to_stdout
+    end
+
+    it "restarts only enabled services with --with-restart flag" do
+      expect(subject.ctl).to receive(:run_sv_command_for_service).with("restart", "bookshelf")
+      expect(subject.ctl).to_not receive(:run_sv_command_for_service).with("restart", "opscode-erchef")
+      expect {
+        subject.run_test_omnibus_command("set-secret", ['bookshelf', 'access_key_id', 'new_key', '--with-restart'])
+      }.to output("Restarting these services: bookshelf\n").to_stdout
+    end
+  end
+
+  #also unlikely that we'd change a secret for a not-enabled service
+  context "when only non-enabled services depend on the changed secret" do
+    before do
+      allow(veil_creds).to receive(:add).with("opscode-reporting", "rabbitmq_password", {:value=>"new_key", :frozen=>true, :force=>true})
+      allow(subject.ctl).to receive(:service_enabled?).with("opscode-reporting").and_return(false)
+    end
+
+    it "does not prompt user to restart services" do
+      expect(subject.ctl).to_not receive(:run_sv_command_for_service)
+      expect {
+        subject.run_test_omnibus_command("set-secret", ['opscode-reporting', 'rabbitmq_password', 'new_key'])
+      }.to_not output.to_stdout
+    end
+
+    it "does not restart services with --with-restart flag" do
+      expect(subject.ctl).to_not receive(:run_sv_command_for_service)
+      expect {
+        subject.run_test_omnibus_command("set-secret", ['opscode-reporting', 'rabbitmq_password', 'new_key', '--with-restart'])
+      }.to_not output.to_stdout
+    end
+  end
+
+  #chef-manage follows a different path as it has several of its own services
+  context "when chef-manage depends on the changed secret and is installed" do
+    before do
+      allow(veil_creds).to receive(:add).with("manage", "secret_key_base", {:value=>"new_key", :frozen=>true, :force=>true})
+      allow(File).to receive(:exist?).with("/opt/chef-manage/sv/").and_return(true)
+    end
+
+    it "prompts user to restart chef-manage" do
+      expect {
+        subject.run_test_omnibus_command("set-secret", ['manage', 'secret_key_base', 'new_key'])
+      }.to output("Please restart these services: chef-manage\n").to_stdout
+    end
+
+    it "restarts chef-manage with --with-restart flag" do
+      expect_any_instance_of(Mixlib::ShellOut).to receive(:run_command)
+      expect {
+        subject.run_test_omnibus_command("set-secret", ['manage', 'secret_key_base', 'new_key', '--with-restart'])
+      }.to output("Restarting these services: chef-manage\n").to_stdout
+    end
+  end
+
+  context "when chef-manage depends on the changed secret but is not installed" do
+    before do
+      allow(veil_creds).to receive(:add).with("manage", "secret_key_base", {:value=>"new_key", :frozen=>true, :force=>true})
+      allow(File).to receive(:exist?).with("/opt/chef-manage/sv/").and_return(false)
```

Comment:
[nit] Should we make `opt/chef-manage/sv` also a constant/variable in the tests?